src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/services/DraftAdjudicationService.kt

```diff
@@ -65,6 +67,28 @@ class DraftAdjudicationService(
     return draftAdjudication.toDto()
   }
 
+  @Transactional
+  fun setOffenceDetails(id: Long, offenceDetails: List<OffenceDetailsDto>): DraftAdjudicationDto {
+    throwIfNoOffenceDetails(offenceDetails)
+
+    val draftAdjudication = draftAdjudicationRepository.findById(id).orElseThrow { throwEntityNotFoundException(id) }
+
+    val newValuesToStore = offenceDetails.map {
+      Offence(
+        offenceCode = it.offenceCode,
+        victimPrisonersNumber = it.victimPrisonersNumber,
+      )
+    }.toMutableList()
+    if (draftAdjudication.offenceDetails != null) {
+      draftAdjudication.offenceDetails!!.removeAll(draftAdjudication.offenceDetails!!)
```

Comment:
This is weird - but it was the only 1-liner I could find to do this.

Reply:
So I might not be understanding this - but are we trying to empty the list out? Can 
draftAdjudication.offenceDetails!!.removeAll(draftAdjudication.offenceDetails!!)
not be replaced by
draftAdjudication.offenceDetails!!.clear()

Reply:
Argh yes.. that does seem obvious. I am not sure, but I might have tried that and had problems... I'll try again for the next PR, which is imminent.

Good call!